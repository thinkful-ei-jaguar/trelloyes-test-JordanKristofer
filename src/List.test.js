import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List.js';


describe('<List />', () => {
  //Smoke Testing 
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <List 
        // header='TEST HEADER'
        cards={[
          {id: 1, title:'TEST 1 Title', content: 'Test is working!'},
          {id: 2, title: 'TEST 2 Title', content: 'Test 2 is wokring!'}
        ]}
      />, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot Testing 
  it('renders this UI as expected', () => {
    //render the component as JSON
    const tree = renderer
      .create(<List 
        cards={[
          {id: 1, title:'TEST 1 Title', content: 'Test is working!'},
          {id: 2, title: 'TEST 2 Title', content: 'Test 2 is wokring!'}
        ]} 
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});

