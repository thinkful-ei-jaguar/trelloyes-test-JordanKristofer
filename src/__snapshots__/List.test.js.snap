// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders this UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        TEST 1 Title
      </h3>
      <p>
        Test is working!
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        TEST 2 Title
      </h3>
      <p>
        Test 2 is wokring!
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
